ui/appui-react/src/appui-react/layout/widget/PreviewHorizontalPanelAlign.tsx

```diff
@@ -0,0 +1,192 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Widget
+ */
+
+import { assert } from "@itwin/core-bentley";
+import { HorizontalAlignment } from "@itwin/core-react";
+import { Button, DropdownMenu, MenuItem } from "@itwin/itwinui-react";
+import * as React from "react";
+import {
+  isHorizontalPanelSide,
+  PanelSideContext,
+} from "../widget-panels/Panel";
+import type {
+  HorizontalPanelSide,
+  PanelSide,
+} from "../widget-panels/PanelTypes";
+
+/** default value used when not provided or disabled */
+const defaultAlignments = {
+  top: HorizontalAlignment.Justify,
+  bottom: HorizontalAlignment.Justify,
+};
+
+const HorizontalPanelAlignContext = React.createContext<{
+  enabled: boolean;
+  alignments: {
+    [side in HorizontalPanelSide]: HorizontalAlignment;
+  };
+  setAlignment: (
+    panel: PanelSide | undefined,
+    alignment: HorizontalAlignment
+  ) => void;
+}>({
+  enabled: false,
+  alignments: defaultAlignments,
+  setAlignment: () => {},
+});
+
+/** Horizontal panel alignment preview feature state.
+ * @internal
+ */
+export function usePreviewHorizontalPanelAlign() {
+  return React.useContext(HorizontalPanelAlignContext);
+}
+
+/** Preview horizontal panel alignment feature provider.
+ * @internal
+ */
+export function PreviewHorizontalPanelAlignFeatureProvider({
+  enabled,
+  children,
+}: {
+  enabled?: boolean;
+  children?: React.ReactNode;
+}) {
+  const [alignments, setAlignments] = React.useState<{
+    [x in HorizontalPanelSide]: HorizontalAlignment;
+  }>({
+    top: HorizontalAlignment.Justify,
+    bottom: HorizontalAlignment.Justify,
+  });
+  const setAlignment = React.useCallback(
+    (panel: PanelSide | undefined, alignment: HorizontalAlignment) => {
+      if (!panel || !isHorizontalPanelSide(panel)) return;
+      setAlignments((prev) => ({
+        ...prev,
+        [panel]: alignment,
+      }));
+    },
+    []
+  );
+  return (
+    <HorizontalPanelAlignContext.Provider
+      value={{
+        enabled: !!enabled,
+        alignments: enabled ? alignments : defaultAlignments,
+        setAlignment,
+      }}
+    >
+      {children}
+    </HorizontalPanelAlignContext.Provider>
+  );
+}
+
+function SvgSide(props: React.SVGProps<SVGSVGElement>) {
+  return (
+    <svg
```

Comment:
@VincentChabot1 is working on new icons for these.

Reply:
Here is the PR
https://github.com/iTwin/iTwinUI-icons/pull/77

Reply:
Do these really need to be in the icons package or can they be part of the appui codebase?

Reply:
For now we could keep this in AppUI repo, until there are other consumers for these icons. My main concern for initial icons was styling and licensing.

Reply:
[DockIcons.zip](https://github.com/iTwin/appui/files/14130173/DockIcons.zip)

Reply:
I replaced the SVGs with the ones in the ZIP.  Raphael is using transforms to get the other 3 icons, if we want to use SVGs instead of transforms, that still needs to be done.

Reply:
![center-bottom](https://github.com/iTwin/appui/assets/849817/734a19fd-b342-4090-b4f7-a9253ef8bd0c)
![center-top](https://github.com/iTwin/appui/assets/849817/d559b9ee-38bb-4cba-8cbd-1dbb033f058d)
![justify-bottom](https://github.com/iTwin/appui/assets/849817/d5e9a5c2-cbe6-4aae-8a7f-4cf55b0186e1)
![justify-top](https://github.com/iTwin/appui/assets/849817/1639ee31-92fd-4b3b-92db-e42fc15e0a2b)
![left-bottom](https://github.com/iTwin/appui/assets/849817/7a2ceb58-af41-4c2b-9f7f-785e0f697f99)
![left-top](https://github.com/iTwin/appui/assets/849817/0ceae25f-783e-46de-a456-ee756d05c491)
![right-bottom](https://github.com/iTwin/appui/assets/849817/dfbc3bb6-5683-49b9-b74a-ab2cfb97c4f2)
![right-top](https://github.com/iTwin/appui/assets/849817/0ba4e56f-013b-4a9f-bb41-2068652d5c7e)

Reply:
Is it by design that they are not squares ?
<img width="82" alt="image" src="https://github.com/iTwin/appui/assets/1904889/d4db34b3-09f1-4011-99b0-345c07661c65">

Reply:
I think we should be fine using the transforms.